extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -252,9 +252,38 @@ function createDicomWebApi(dicomWebConfig, UserAuthenticationService) {
         sortFunction
       );
 
+      /**
+       * naturalizes the dataset, and adds a retrieve bulkdata method
+       * to any values containing BulkDataURI.
+       * @param {*} instance
+       * @returns naturalized dataset, with retrieveBulkData methods
+       */
+      const addRetrieveBulkData = instance => {
+        const naturalized = naturalizeDataset(instance);
+        Object.keys(naturalized).forEach(key => {
+          const value = naturalized[key];
+          if (value && value.BulkDataURI && !value.Value) {
+            // Provide a method to fetch this
+            value.retrieveBulkData = () => {
+              const options = {
+                multipart: false,
+                BulkDataURI: value.BulkDataURI,
+                StudyInstanceUID: naturalized.StudyInstanceUID,
+              };
+              return qidoDicomWebClient.retrieveBulkData(options).then(val => {
+                const ret = val && val[0] || undefined;
+                value.Value = ret;
+                return ret;
+              });
```

Comment:
@chafey  - here is the corresponding change in OHIFv3 - the retrieveBulkData call gets made and returns a promise to return an array buffer.  Fairly simple to implement, and is completely backwards compatible with OHIF v2, and any other cornerstone clients.